//
// This file was generated by the JavaTM Architecture for XML Binding(JAXB) Reference Implementation, v2.2.4 
// See <a href="http://java.sun.com/xml/jaxb">http://java.sun.com/xml/jaxb</a> 
// Any modifications to this file will be lost upon recompilation of the source schema. 
// Generated on: 2013.06.15 at 10:06:18 PM CEST 
//


package ncbisegout;

import javax.xml.bind.JAXBElement;
import javax.xml.bind.annotation.XmlElementDecl;
import javax.xml.bind.annotation.XmlRegistry;
import javax.xml.namespace.QName;


/**
 * This object contains factory methods for each 
 * Java content interface and Java element interface 
 * generated in the ncbisegout package. 
 * <p>An ObjectFactory allows you to programatically 
 * construct new instances of the Java representation 
 * for XML content. The Java representation of XML 
 * content can consist of schema derived interfaces 
 * and classes representing the binding of schema 
 * type definitions, element declarations and model 
 * groups.  Factory methods for each of these are 
 * provided in this class.
 * 
 */
@XmlRegistry
public class ObjectFactory {

    private final static QName _LowComplexityRegion_QNAME = new QName("http://i12r-tbl.informatik.tu-muenchen.de/~jonas/ncbiseq/output", "lowComplexityRegion");
    private final static QName _FeatureTypeClassConcept_QNAME = new QName("http://bioxsd.org/BioXSD-1.1", "classConcept");
    private final static QName _FeatureTypeClassName_QNAME = new QName("http://bioxsd.org/BioXSD-1.1", "className");
    private final static QName _FeatureTypeProperties_QNAME = new QName("http://bioxsd.org/BioXSD-1.1", "properties");
    private final static QName _FeatureTypeEqualAlias_QNAME = new QName("http://bioxsd.org/BioXSD-1.1", "equalAlias");
    private final static QName _FeatureTypeName_QNAME = new QName("http://bioxsd.org/BioXSD-1.1", "name");
    private final static QName _FeatureTypeEqualConcept_QNAME = new QName("http://bioxsd.org/BioXSD-1.1", "equalConcept");

    /**
     * Create a new ObjectFactory that can be used to create new instances of schema derived classes for package: ncbisegout
     * 
     */
    public ObjectFactory() {
    }

    /**
     * Create an instance of {@link FeatureOccurrenceData }
     * 
     */
    public FeatureOccurrenceData createFeatureOccurrenceData() {
        return new FeatureOccurrenceData();
    }

    /**
     * Create an instance of {@link FeatureOccurrenceData.Alignment }
     * 
     */
    public FeatureOccurrenceData.Alignment createFeatureOccurrenceDataAlignment() {
        return new FeatureOccurrenceData.Alignment();
    }

    /**
     * Create an instance of {@link GeneticCode }
     * 
     */
    public GeneticCode createGeneticCode() {
        return new GeneticCode();
    }

    /**
     * Create an instance of {@link FeatureTypeDetails }
     * 
     */
    public FeatureTypeDetails createFeatureTypeDetails() {
        return new FeatureTypeDetails();
    }

    /**
     * Create an instance of {@link MethodResult }
     * 
     */
    public MethodResult createMethodResult() {
        return new MethodResult();
    }

    /**
     * Create an instance of {@link FeatureRecord }
     * 
     */
    public FeatureRecord createFeatureRecord() {
        return new FeatureRecord();
    }

    /**
     * Create an instance of {@link FeatureRecord.BlockWithOccurrenceReferences }
     * 
     */
    public FeatureRecord.BlockWithOccurrenceReferences createFeatureRecordBlockWithOccurrenceReferences() {
        return new FeatureRecord.BlockWithOccurrenceReferences();
    }

    /**
     * Create an instance of {@link FeatureRecord.BlockWithOccurrenceReferences.Annotation }
     * 
     */
    public FeatureRecord.BlockWithOccurrenceReferences.Annotation createFeatureRecordBlockWithOccurrenceReferencesAnnotation() {
        return new FeatureRecord.BlockWithOccurrenceReferences.Annotation();
    }

    /**
     * Create an instance of {@link FeatureRecord.BlockWithOccurrenceReferences.Annotation.Occurrence }
     * 
     */
    public FeatureRecord.BlockWithOccurrenceReferences.Annotation.Occurrence createFeatureRecordBlockWithOccurrenceReferencesAnnotationOccurrence() {
        return new FeatureRecord.BlockWithOccurrenceReferences.Annotation.Occurrence();
    }

    /**
     * Create an instance of {@link FeatureRecord.BlockWithOccurrenceReferences.Annotation.Occurrence.Association }
     * 
     */
    public FeatureRecord.BlockWithOccurrenceReferences.Annotation.Occurrence.Association createFeatureRecordBlockWithOccurrenceReferencesAnnotationOccurrenceAssociation() {
        return new FeatureRecord.BlockWithOccurrenceReferences.Annotation.Occurrence.Association();
    }

    /**
     * Create an instance of {@link FeatureRecord.Annotation }
     * 
     */
    public FeatureRecord.Annotation createFeatureRecordAnnotation() {
        return new FeatureRecord.Annotation();
    }

    /**
     * Create an instance of {@link Insert }
     * 
     */
    public Insert createInsert() {
        return new Insert();
    }

    /**
     * Create an instance of {@link AminoacidSequenceRecord }
     * 
     */
    public AminoacidSequenceRecord createAminoacidSequenceRecord() {
        return new AminoacidSequenceRecord();
    }

    /**
     * Create an instance of {@link ncbisegout.Method }
     * 
     */
    public ncbisegout.Method createMethod() {
        return new ncbisegout.Method();
    }

    /**
     * Create an instance of {@link TranslationData }
     * 
     */
    public TranslationData createTranslationData() {
        return new TranslationData();
    }

    /**
     * Create an instance of {@link GeneralAminoacidSequenceRecord }
     * 
     */
    public GeneralAminoacidSequenceRecord createGeneralAminoacidSequenceRecord() {
        return new GeneralAminoacidSequenceRecord();
    }

    /**
     * Create an instance of {@link FeatureType }
     * 
     */
    public FeatureType createFeatureType() {
        return new FeatureType();
    }

    /**
     * Create an instance of {@link SequencePosition }
     * 
     */
    public SequencePosition createSequencePosition() {
        return new SequencePosition();
    }

    /**
     * Create an instance of {@link OutsideSequenceSegment }
     * 
     */
    public OutsideSequenceSegment createOutsideSequenceSegment() {
        return new OutsideSequenceSegment();
    }

    /**
     * Create an instance of {@link Score }
     * 
     */
    public Score createScore() {
        return new Score();
    }

    /**
     * Create an instance of {@link Result }
     * 
     */
    public Result createResult() {
        return new Result();
    }

    /**
     * Create an instance of {@link AminoacidTranslationData }
     * 
     */
    public AminoacidTranslationData createAminoacidTranslationData() {
        return new AminoacidTranslationData();
    }

    /**
     * Create an instance of {@link GeneralSequencePoint }
     * 
     */
    public GeneralSequencePoint createGeneralSequencePoint() {
        return new GeneralSequencePoint();
    }

    /**
     * Create an instance of {@link GeneralOutsideSequencePosition }
     * 
     */
    public GeneralOutsideSequencePosition createGeneralOutsideSequencePosition() {
        return new GeneralOutsideSequencePosition();
    }

    /**
     * Create an instance of {@link GeneralOutsideSequenceSegment }
     * 
     */
    public GeneralOutsideSequenceSegment createGeneralOutsideSequenceSegment() {
        return new GeneralOutsideSequenceSegment();
    }

    /**
     * Create an instance of {@link SequenceReference }
     * 
     */
    public SequenceReference createSequenceReference() {
        return new SequenceReference();
    }

    /**
     * Create an instance of {@link EntryReference }
     * 
     */
    public EntryReference createEntryReference() {
        return new EntryReference();
    }

    /**
     * Create an instance of {@link OntologyReference }
     * 
     */
    public OntologyReference createOntologyReference() {
        return new OntologyReference();
    }

    /**
     * Create an instance of {@link GeneralOutsideSequencePoint }
     * 
     */
    public GeneralOutsideSequencePoint createGeneralOutsideSequencePoint() {
        return new GeneralOutsideSequencePoint();
    }

    /**
     * Create an instance of {@link AminoacidSequenceAlignment }
     * 
     */
    public AminoacidSequenceAlignment createAminoacidSequenceAlignment() {
        return new AminoacidSequenceAlignment();
    }

    /**
     * Create an instance of {@link SequenceSegment }
     * 
     */
    public SequenceSegment createSequenceSegment() {
        return new SequenceSegment();
    }

    /**
     * Create an instance of {@link GeneralNucleotideSequenceAlignment }
     * 
     */
    public GeneralNucleotideSequenceAlignment createGeneralNucleotideSequenceAlignment() {
        return new GeneralNucleotideSequenceAlignment();
    }

    /**
     * Create an instance of {@link AlignedGeneralNucleotideSequence }
     * 
     */
    public AlignedGeneralNucleotideSequence createAlignedGeneralNucleotideSequence() {
        return new AlignedGeneralNucleotideSequence();
    }

    /**
     * Create an instance of {@link ExperimentalEvidence }
     * 
     */
    public ExperimentalEvidence createExperimentalEvidence() {
        return new ExperimentalEvidence();
    }

    /**
     * Create an instance of {@link Frameshift }
     * 
     */
    public Frameshift createFrameshift() {
        return new Frameshift();
    }

    /**
     * Create an instance of {@link DatabaseReference }
     * 
     */
    public DatabaseReference createDatabaseReference() {
        return new DatabaseReference();
    }

    /**
     * Create an instance of {@link Flag }
     * 
     */
    public Flag createFlag() {
        return new Flag();
    }

    /**
     * Create an instance of {@link SequencePartition }
     * 
     */
    public SequencePartition createSequencePartition() {
        return new SequencePartition();
    }

    /**
     * Create an instance of {@link GeneralSequenceSegment }
     * 
     */
    public GeneralSequenceSegment createGeneralSequenceSegment() {
        return new GeneralSequenceSegment();
    }

    /**
     * Create an instance of {@link ncbisegout.SemanticConcept }
     * 
     */
    public ncbisegout.SemanticConcept createSemanticConcept() {
        return new ncbisegout.SemanticConcept();
    }

    /**
     * Create an instance of {@link AlignedNucleotideSequence }
     * 
     */
    public AlignedNucleotideSequence createAlignedNucleotideSequence() {
        return new AlignedNucleotideSequence();
    }

    /**
     * Create an instance of {@link BiosequenceAlignment }
     * 
     */
    public BiosequenceAlignment createBiosequenceAlignment() {
        return new BiosequenceAlignment();
    }

    /**
     * Create an instance of {@link OutsideSequencePoint }
     * 
     */
    public OutsideSequencePoint createOutsideSequencePoint() {
        return new OutsideSequencePoint();
    }

    /**
     * Create an instance of {@link GeneralAminoacidSequenceAlignment }
     * 
     */
    public GeneralAminoacidSequenceAlignment createGeneralAminoacidSequenceAlignment() {
        return new GeneralAminoacidSequenceAlignment();
    }

    /**
     * Create an instance of {@link GeneralNucleotideSequenceRecord }
     * 
     */
    public GeneralNucleotideSequenceRecord createGeneralNucleotideSequenceRecord() {
        return new GeneralNucleotideSequenceRecord();
    }

    /**
     * Create an instance of {@link BiosequenceRecord }
     * 
     */
    public BiosequenceRecord createBiosequenceRecord() {
        return new BiosequenceRecord();
    }

    /**
     * Create an instance of {@link GeneralSequenceInsertionPoint }
     * 
     */
    public GeneralSequenceInsertionPoint createGeneralSequenceInsertionPoint() {
        return new GeneralSequenceInsertionPoint();
    }

    /**
     * Create an instance of {@link GeneralSequencePosition }
     * 
     */
    public GeneralSequencePosition createGeneralSequencePosition() {
        return new GeneralSequencePosition();
    }

    /**
     * Create an instance of {@link NucleotideSequenceRecord }
     * 
     */
    public NucleotideSequenceRecord createNucleotideSequenceRecord() {
        return new NucleotideSequenceRecord();
    }

    /**
     * Create an instance of {@link Species }
     * 
     */
    public Species createSpecies() {
        return new Species();
    }

    /**
     * Create an instance of {@link SequenceInsertionPoint }
     * 
     */
    public SequenceInsertionPoint createSequenceInsertionPoint() {
        return new SequenceInsertionPoint();
    }

    /**
     * Create an instance of {@link AlignedBiosequence }
     * 
     */
    public AlignedBiosequence createAlignedBiosequence() {
        return new AlignedBiosequence();
    }

    /**
     * Create an instance of {@link AlignedGeneralAminoacidSequence }
     * 
     */
    public AlignedGeneralAminoacidSequence createAlignedGeneralAminoacidSequence() {
        return new AlignedGeneralAminoacidSequence();
    }

    /**
     * Create an instance of {@link ncbisegout.ScoreType }
     * 
     */
    public ncbisegout.ScoreType createScoreType() {
        return new ncbisegout.ScoreType();
    }

    /**
     * Create an instance of {@link NucleotideSequenceAlignment }
     * 
     */
    public NucleotideSequenceAlignment createNucleotideSequenceAlignment() {
        return new NucleotideSequenceAlignment();
    }

    /**
     * Create an instance of {@link AlignedAminoacidSequence }
     * 
     */
    public AlignedAminoacidSequence createAlignedAminoacidSequence() {
        return new AlignedAminoacidSequence();
    }

    /**
     * Create an instance of {@link Gap }
     * 
     */
    public Gap createGap() {
        return new Gap();
    }

    /**
     * Create an instance of {@link PredictionResult }
     * 
     */
    public PredictionResult createPredictionResult() {
        return new PredictionResult();
    }

    /**
     * Create an instance of {@link OutsideSequencePosition }
     * 
     */
    public OutsideSequencePosition createOutsideSequencePosition() {
        return new OutsideSequencePosition();
    }

    /**
     * Create an instance of {@link GeneralSequencePartition }
     * 
     */
    public GeneralSequencePartition createGeneralSequencePartition() {
        return new GeneralSequencePartition();
    }

    /**
     * Create an instance of {@link SequencePoint }
     * 
     */
    public SequencePoint createSequencePoint() {
        return new SequencePoint();
    }

    /**
     * Create an instance of {@link Evidence }
     * 
     */
    public Evidence createEvidence() {
        return new Evidence();
    }

    /**
     * Create an instance of {@link FeatureOccurrenceData.Variation }
     * 
     */
    public FeatureOccurrenceData.Variation createFeatureOccurrenceDataVariation() {
        return new FeatureOccurrenceData.Variation();
    }

    /**
     * Create an instance of {@link FeatureOccurrenceData.Alignment.AlignmentOfReference }
     * 
     */
    public FeatureOccurrenceData.Alignment.AlignmentOfReference createFeatureOccurrenceDataAlignmentAlignmentOfReference() {
        return new FeatureOccurrenceData.Alignment.AlignmentOfReference();
    }

    /**
     * Create an instance of {@link GeneticCode.Codon }
     * 
     */
    public GeneticCode.Codon createGeneticCodeCodon() {
        return new GeneticCode.Codon();
    }

    /**
     * Create an instance of {@link FeatureTypeDetails.Association }
     * 
     */
    public FeatureTypeDetails.Association createFeatureTypeDetailsAssociation() {
        return new FeatureTypeDetails.Association();
    }

    /**
     * Create an instance of {@link MethodResult.Method }
     * 
     */
    public MethodResult.Method createMethodResultMethod() {
        return new MethodResult.Method();
    }

    /**
     * Create an instance of {@link FeatureRecord.ReferenceSequence }
     * 
     */
    public FeatureRecord.ReferenceSequence createFeatureRecordReferenceSequence() {
        return new FeatureRecord.ReferenceSequence();
    }

    /**
     * Create an instance of {@link FeatureRecord.BlockWithOccurrenceReferences.Database }
     * 
     */
    public FeatureRecord.BlockWithOccurrenceReferences.Database createFeatureRecordBlockWithOccurrenceReferencesDatabase() {
        return new FeatureRecord.BlockWithOccurrenceReferences.Database();
    }

    /**
     * Create an instance of {@link FeatureRecord.BlockWithOccurrenceReferences.Ontology }
     * 
     */
    public FeatureRecord.BlockWithOccurrenceReferences.Ontology createFeatureRecordBlockWithOccurrenceReferencesOntology() {
        return new FeatureRecord.BlockWithOccurrenceReferences.Ontology();
    }

    /**
     * Create an instance of {@link FeatureRecord.BlockWithOccurrenceReferences.SemanticConcept }
     * 
     */
    public FeatureRecord.BlockWithOccurrenceReferences.SemanticConcept createFeatureRecordBlockWithOccurrenceReferencesSemanticConcept() {
        return new FeatureRecord.BlockWithOccurrenceReferences.SemanticConcept();
    }

    /**
     * Create an instance of {@link FeatureRecord.BlockWithOccurrenceReferences.Method }
     * 
     */
    public FeatureRecord.BlockWithOccurrenceReferences.Method createFeatureRecordBlockWithOccurrenceReferencesMethod() {
        return new FeatureRecord.BlockWithOccurrenceReferences.Method();
    }

    /**
     * Create an instance of {@link FeatureRecord.BlockWithOccurrenceReferences.ScoreType }
     * 
     */
    public FeatureRecord.BlockWithOccurrenceReferences.ScoreType createFeatureRecordBlockWithOccurrenceReferencesScoreType() {
        return new FeatureRecord.BlockWithOccurrenceReferences.ScoreType();
    }

    /**
     * Create an instance of {@link FeatureRecord.BlockWithOccurrenceReferences.Annotation.CondensedReferences }
     * 
     */
    public FeatureRecord.BlockWithOccurrenceReferences.Annotation.CondensedReferences createFeatureRecordBlockWithOccurrenceReferencesAnnotationCondensedReferences() {
        return new FeatureRecord.BlockWithOccurrenceReferences.Annotation.CondensedReferences();
    }

    /**
     * Create an instance of {@link FeatureRecord.BlockWithOccurrenceReferences.Annotation.Occurrence.Association.AssociatedFeatureOccurrence }
     * 
     */
    public FeatureRecord.BlockWithOccurrenceReferences.Annotation.Occurrence.Association.AssociatedFeatureOccurrence createFeatureRecordBlockWithOccurrenceReferencesAnnotationOccurrenceAssociationAssociatedFeatureOccurrence() {
        return new FeatureRecord.BlockWithOccurrenceReferences.Annotation.Occurrence.Association.AssociatedFeatureOccurrence();
    }

    /**
     * Create an instance of {@link FeatureRecord.Annotation.CondensedReferences }
     * 
     */
    public FeatureRecord.Annotation.CondensedReferences createFeatureRecordAnnotationCondensedReferences() {
        return new FeatureRecord.Annotation.CondensedReferences();
    }

    /**
     * Create an instance of {@link JAXBElement }{@code <}{@link FeatureRecord }{@code >}}
     * 
     */
    @XmlElementDecl(namespace = "http://i12r-tbl.informatik.tu-muenchen.de/~jonas/ncbiseq/output", name = "lowComplexityRegion")
    public JAXBElement<FeatureRecord> createLowComplexityRegion(FeatureRecord value) {
        return new JAXBElement<FeatureRecord>(_LowComplexityRegion_QNAME, FeatureRecord.class, null, value);
    }

    /**
     * Create an instance of {@link JAXBElement }{@code <}{@link ncbisegout.SemanticConcept }{@code >}}
     * 
     */
    @XmlElementDecl(namespace = "http://bioxsd.org/BioXSD-1.1", name = "classConcept", scope = FeatureType.class)
    public JAXBElement<ncbisegout.SemanticConcept> createFeatureTypeClassConcept(ncbisegout.SemanticConcept value) {
        return new JAXBElement<ncbisegout.SemanticConcept>(_FeatureTypeClassConcept_QNAME, ncbisegout.SemanticConcept.class, FeatureType.class, value);
    }

    /**
     * Create an instance of {@link JAXBElement }{@code <}{@link String }{@code >}}
     * 
     */
    @XmlElementDecl(namespace = "http://bioxsd.org/BioXSD-1.1", name = "className", scope = FeatureType.class)
    public JAXBElement<String> createFeatureTypeClassName(String value) {
        return new JAXBElement<String>(_FeatureTypeClassName_QNAME, String.class, FeatureType.class, value);
    }

    /**
     * Create an instance of {@link JAXBElement }{@code <}{@link FeatureTypeDetails }{@code >}}
     * 
     */
    @XmlElementDecl(namespace = "http://bioxsd.org/BioXSD-1.1", name = "properties", scope = FeatureType.class)
    public JAXBElement<FeatureTypeDetails> createFeatureTypeProperties(FeatureTypeDetails value) {
        return new JAXBElement<FeatureTypeDetails>(_FeatureTypeProperties_QNAME, FeatureTypeDetails.class, FeatureType.class, value);
    }

    /**
     * Create an instance of {@link JAXBElement }{@code <}{@link String }{@code >}}
     * 
     */
    @XmlElementDecl(namespace = "http://bioxsd.org/BioXSD-1.1", name = "equalAlias", scope = FeatureType.class)
    public JAXBElement<String> createFeatureTypeEqualAlias(String value) {
        return new JAXBElement<String>(_FeatureTypeEqualAlias_QNAME, String.class, FeatureType.class, value);
    }

    /**
     * Create an instance of {@link JAXBElement }{@code <}{@link String }{@code >}}
     * 
     */
    @XmlElementDecl(namespace = "http://bioxsd.org/BioXSD-1.1", name = "name", scope = FeatureType.class)
    public JAXBElement<String> createFeatureTypeName(String value) {
        return new JAXBElement<String>(_FeatureTypeName_QNAME, String.class, FeatureType.class, value);
    }

    /**
     * Create an instance of {@link JAXBElement }{@code <}{@link ncbisegout.SemanticConcept }{@code >}}
     * 
     */
    @XmlElementDecl(namespace = "http://bioxsd.org/BioXSD-1.1", name = "equalConcept", scope = FeatureType.class)
    public JAXBElement<ncbisegout.SemanticConcept> createFeatureTypeEqualConcept(ncbisegout.SemanticConcept value) {
        return new JAXBElement<ncbisegout.SemanticConcept>(_FeatureTypeEqualConcept_QNAME, ncbisegout.SemanticConcept.class, FeatureType.class, value);
    }

}
